tests/string_array/test_pandas_extension.py

```diff
@@ -15,54 +18,115 @@
     BaseSetitemTests,
 )
 
-from fletcher import StringArray, StringDtype
+from fletcher import StringArray, FletcherDtype, Date64Array
+
+FletcherTestType = namedtuple(
+    "FletcherTestType",
+    [
+        "dtype",
+        "data",
+        "data_missing",
+        "data_for_grouping",
+        "data_for_sorting",
+        "data_missing_for_sorting",
+    ],
+)
+
+test_types = [
```

Comment:
Unfortunately, I do not have a better idea, yet, but this looks like it's going to blow up to an unreasonable level if we have to manage a huge list with all possible dtypes. For now, this is fine I guess